FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/ApiClientRepository.kt

```diff
@@ -0,0 +1,387 @@
+/*
+ * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
+ * Contributors: DenBond7
+ */
+
+package com.flowcrypt.email.api.retrofit
+
+import android.content.Context
+import com.flowcrypt.email.Constants
+import com.flowcrypt.email.R
+import com.flowcrypt.email.api.retrofit.base.BaseApiRepository
+import com.flowcrypt.email.api.retrofit.request.model.MessageUploadRequest
+import com.flowcrypt.email.api.retrofit.request.model.PostHelpFeedbackModel
+import com.flowcrypt.email.api.retrofit.request.model.WelcomeMessageModel
+import com.flowcrypt.email.api.retrofit.response.api.ClientConfigurationResponse
+import com.flowcrypt.email.api.retrofit.response.api.EkmPrivateKeysResponse
+import com.flowcrypt.email.api.retrofit.response.api.FesServerResponse
+import com.flowcrypt.email.api.retrofit.response.api.MessageReplyTokenResponse
+import com.flowcrypt.email.api.retrofit.response.api.MessageUploadResponse
+import com.flowcrypt.email.api.retrofit.response.api.PostHelpFeedbackResponse
+import com.flowcrypt.email.api.retrofit.response.attester.PubResponse
+import com.flowcrypt.email.api.retrofit.response.attester.SubmitPubKeyResponse
+import com.flowcrypt.email.api.retrofit.response.attester.WelcomeMessageResponse
+import com.flowcrypt.email.api.retrofit.response.base.Result
+import com.flowcrypt.email.api.retrofit.response.model.ClientConfiguration
+import com.flowcrypt.email.api.retrofit.response.oauth2.MicrosoftOAuth2TokenResponse
+import com.flowcrypt.email.api.wkd.WkdClient
+import com.flowcrypt.email.extensions.kotlin.isValidEmail
+import com.flowcrypt.email.extensions.org.bouncycastle.openpgp.armor
+import com.google.gson.GsonBuilder
+import com.google.gson.JsonObject
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.MultipartBody
+import okhttp3.OkHttpClient
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.ResponseBody.Companion.toResponseBody
+import org.pgpainless.algorithm.EncryptionPurpose
+import org.pgpainless.key.info.KeyRingInfo
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.net.HttpURLConnection
+import java.util.concurrent.TimeUnit
+
+/**
+ * This class describes all network requests over the app.
+ *
+ * @author Denis Bondarenko
+ *         Date: 10/24/19
+ *         Time: 6:10 PM
+ *         E-mail: DenBond7@gmail.com
+ */
+object ApiClientRepository : BaseApiRepository {
+  object FES : BaseApiRepository {
+    /**
+     * @param context Interface to global information about an application environment.
+     * @param baseFesUrlPath A base FES URL path.
+     * @param domain A company domain.
+     * @param idToken OIDC token.
+     */
+    suspend fun getClientConfigurationFromFes(
```

Comment:
Since now it's part of the FES object, can rename this to `getClientConfiguration`